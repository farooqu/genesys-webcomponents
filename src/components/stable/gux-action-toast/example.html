<h1>gux-action-toast</h1>

<gux-action-toast>
  <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
  <div slot="title">Title</div>
  <div slot="message">This is the message</div>
  <gux-button slot="positive-button" accent="primary">Accept</gux-button>
  <gux-button slot="negative-button">Reject</gux-button>
</gux-action-toast>

<gux-action-toast>
  <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
  <div slot="title">Title</div>
  <dl slot="message">
    <dt>Name</dt>
    <dd>Clarissa Hopkirk</dd>
    <dt>Topic</dt>
    <dd>Billing Issue</dd>
    <dt>Tier</dt>
    <dd>Gold</dd>
  </dl>
  <gux-button slot="positive-button" accent="primary">Accept</gux-button>
  <gux-button slot="negative-button">Reject</gux-button>
</gux-action-toast>
