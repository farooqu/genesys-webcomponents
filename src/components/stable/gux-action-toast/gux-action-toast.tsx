import { Component, h, Element, Host, JSX } from '@stencil/core';
import { trackComponent } from '../../../usage-tracking';

/**
 * @slot icon - Required slot for gux-icon
 * @slot title - Required slot for the action toast title
 * @slot message - Required slot for the action toast message
 * @slot negative-button - Required slot for the action toast negative button
 * @slot positive-button - Required slot for the action toast positive button
 */
@Component({
  styleUrl: 'gux-action-toast.less',
  tag: 'gux-action-toast',
  shadow: true
})
export class GuxActionToast {
  @Element()
  private root: HTMLElement;

  componentWillLoad(): void {
    trackComponent(this.root);
  }

  render(): JSX.Element {
    return (
      <Host>
        <div class="gux-header">
          <div class="gux-icon">
            <slot name="icon" />
          </div>

          <div class="gux-title">
            <slot name="title" />
          </div>
        </div>

        <div class="gux-message">
          <slot name="message" />
        </div>

        <div class="gux-action-buttons">
          <div class="gux-positive-button">
            <slot name="positive-button" />
          </div>

          <div class="gux-negative-button">
            <slot name="negative-button" />
          </div>
        </div>
      </Host>
    ) as JSX.Element;
  }
}
