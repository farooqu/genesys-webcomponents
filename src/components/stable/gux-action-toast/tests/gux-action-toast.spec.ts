import { newSpecPage } from '@stencil/core/testing';
import { GuxActionToast } from '../gux-action-toast';

const components = [GuxActionToast];
const language = 'en';

describe('gux-action-toast', () => {
  describe('#render', () => {
    [
      {
        description: 'should render action toast',
        html: `
          <gux-action-toast accent="neutral">
            <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
            <div slot="title">Title</div>
            <div slot="message">This is the message</div>
            <gux-button slot="negitive-button">Reject</gux-button>
            <gux-button slot="positive-button" accent="primary">Accept</gux-button>
          </gux-action-toast>
        `
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.rootInstance).toBeInstanceOf(GuxActionToast);

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
