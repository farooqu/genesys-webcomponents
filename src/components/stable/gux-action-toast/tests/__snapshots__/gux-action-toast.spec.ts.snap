// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-action-toast #render should render action toast 1`] = `
<gux-action-toast accent="neutral">
  <!---->
  <div class="gux-header">
    <div class="gux-icon">
      <gux-icon decorative="" icon-name="user-add" slot="icon"></gux-icon>
    </div>
    <div class="gux-title">
      <div slot="title">
        Title
      </div>
    </div>
  </div>
  <div class="gux-message">
    <div slot="message">
      This is the message
    </div>
  </div>
  <div class="gux-action-buttons">
    <div class="gux-negitive-button">
      <gux-button slot="negitive-button">
        Reject
      </gux-button>
    </div>
    <div class="gux-positive-button">
      <gux-button accent="primary" slot="positive-button">
        Accept
      </gux-button>
    </div>
  </div>
</gux-action-toast>
`;
